maths/logarithm.py

```diff
@@ -0,0 +1,79 @@
+"""
+https://en.wikipedia.org/wiki/Logarithm
+"""
+
+
+def logarithm(end: float, base: float, err: float = 0.000001) -> float:
+    """
+    Return the logarithm with base base of x
+    :param end: the number
+    :param base: the base of the exponential
+    :return:
+    
+    >>> logarithm(2, 9)
+    0.31546499999999994
+    >>> logarithm(9, 2)
+    3.1699250000000005
+    >>> logarithm(0.2, 0.9)
+    15.275499999999994
+    >>> logarithm(0.9, 0.2)
+    0.065464
+    >>> logarithm(0.2, 9)
+    -0.7324849999999997
+    >>> logarithm(2, 0.9)
+    -6.578809999999997
+    >>> logarithm(9, 0.2)
+    -1.3652123999999999
+    >>> logarithm(0.9, 2)
+    -0.15200200000000003
+    """
+    
+    if base < 0 or end < 0:
+        raise ValueError('logarithm() takes 2 positive values')

```

Comment:
Please add tests where base is 0 and base is -1 so we can see how this function fails.